app/web/features/userQueries/useLiteUsers.test.tsx

```diff
@@ -0,0 +1,283 @@
+          isLoading: false,
+          isRefetching: false,
+        })
+      );
+    });
+  });
+
+  describe("useLiteUser (singular)", () => {
+    const mockGetLiteUser = service.user.getLiteUser as jest.Mock;
+
+    it("Should return loading state correctly", async () => {
+      mockGetLiteUser.mockResolvedValue(getLiteUser("1"));
+
+      const { result, waitFor } = renderHook(() => useLiteUser(1), { wrapper });
+
+      expect(result.current.isLoading).toBe(true);
+      expect(result.current.data).toBeUndefined();
+
+      await waitFor(() => expect(result.current.isLoading).toBe(false));
+    });
+
+    it("Should return user when loading is complete", async () => {
+      mockGetLiteUser.mockResolvedValue(getLiteUser("2"));
+
+      const { result, waitForNextUpdate } = renderHook(() => useLiteUser(1), {
+        wrapper,
+      });
+
+      await waitForNextUpdate();
+      expect(mockGetLiteUser).toHaveBeenCalledTimes(1);
+
+      expect(result.current).toEqual(
+        expect.objectContaining({
+          data: mockLiteUsers[1],
+          error: null,
+          isError: false,
+          isFetching: false,
+          isLoading: false,
+          isRefetching: false,
+        })
+      );
+    });
+
+    it("Should return an error if the query fails", async () => {
+      mockConsoleError();
+      const error = new RpcError(500, "Some error", {});
+      mockGetLiteUser.mockRejectedValue(error);
+
+      const { result, waitForNextUpdate } = renderHook(() => useLiteUser(1), {
+        wrapper,
+      });
+
+      await waitForNextUpdate();
+
+      expect(result.current.isLoading).toBe(false);
+      expect(result.current.error).toBe(error);
+      expect(result.current.data).toBeUndefined();
+    });
+
+    it("Should not run the query function when id is undefined", () => {
+      // @ts-expect-error
+      renderHook(() => useLiteUser(), { wrapper });
+
+      expect(mockGetLiteUser).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("useLiteUser (singular) - cached data", () => {
+    const mockGetLiteUser = service.user.getLiteUser as jest.Mock;
+
+    const sharedClient = new QueryClient({
+      defaultOptions: { queries: { retry: false } },
+    });
+
+    const sharedClientWrapper = ({
+      children,
+    }: {
+      children: React.ReactNode;
+    }) => (
+      <QueryClientProvider client={sharedClient}>
+        {children}
+      </QueryClientProvider>
+    );
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+
+      sharedClient.clear();
+      // Set the cached data for the specific liteUserId
+      sharedClient.setQueryData(liteUserKey(3), getLiteUser("3"));
+    });
+
+    it("Uses cached data instead of refetching", async () => {
+      const { result } = renderHook(() => useLiteUser(3), {
+        wrapper: sharedClientWrapper,
+      });
+
+      expect(mockGetLiteUser).not.toHaveBeenCalled();
+      expect(await sharedClient.getQueryData(liteUserKey(3))).toEqual(
+        mockLiteUsers[2]
+      );
```

Comment:
I think it'd be better to assert on `result.current.data` to prove the hook works here rather than interacting on the `sharedClient`, since we just set the query data in the `beforeEach` so I'd expect to have data there. We're testing React Query is doing its job here instead of the hook otherwise?

Reply:
Fixed!